### PR TITLE
Support table cells with dict values

### PR DESCRIPTION
**Pull request by xbrianh**
depends on #273

### CODE REVIEW

File: terra_notebook_utils/table.py [hunk 1]
```diff
@@ -62,14 +62,19 @@ def _init_request_data(self):
...
         request_data = list()
+        types: Set[type]
         for name, val in row.attributes.items():
             if isinstance(val, str):
                 update_ops = list()  # No Firecloud update operations needed for string values
             elif isinstance(val, (int, float, bool)):
                 update_ops = [dict(op="AddUpdateAttribute", attributeName=name, addUpdateAttribute=val)]
+            elif hasattr(val, "keys") and hasattr(val, "items"):  # TODO: better way to detect dict-like objects?
+                types = {type(k) for k in val.keys()}
+                assert 1 == len(types) and str == types.pop()
```
Comment (DailyDreaming) [inline terra_notebook_utils/table.py:73]: Is `assert types == {str}` equivalent here?

File: terra_notebook_utils/table.py [hunk 2]
```diff
@@ -62,14 +62,19 @@ def _init_request_data(self):
 
     def _get_row_update_request_data(self, row: Row) -> List[Dict[str, Any]]:
         request_data = list()
+        types: Set[type]
         for name, val in row.attributes.items():
             if isinstance(val, str):
                 update_ops = list()  # No Firecloud update operations needed for string values
             elif isinstance(val, (int, float, bool)):
                 update_ops = [dict(op="AddUpdateAttribute", attributeName=name, addUpdateAttribute=val)]
+            elif hasattr(val, "keys") and hasattr(val, "items"):  # TODO: better way to detect dict-like objects?
```
Comment (DailyDreaming) [inline terra_notebook_utils/table.py:71]: I believe that `collections.Mapping` would probably work here (e.g. `isinstance(val, Mapping)`).  Dictionaries type as `collections.Mapping` and it seems to define both items and keys: https://docs.python.org/3.1/library/collections.html#collections.Mapping

**Review by DailyDreaming — APPROVED** [review]
LGTM.  See comments.

File: terra_notebook_utils/table.py (hunk 2 above)
Comment (DailyDreaming) [reply terra_notebook_utils/table.py:71]: Edit: This is `collections.abc` now apparently: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

File: terra_notebook_utils/table.py (hunk 1 above)
Comment (xbrianh) [reply terra_notebook_utils/table.py]: Yes, they’re equivalent. I’ll go with your suggestion, which looks cleaner.

File: terra_notebook_utils/table.py (hunk 2 above)
Comment (xbrianh) [reply terra_notebook_utils/table.py]: Good suggestion! I went with [typing.Mapping](https://docs.python.org/3/library/typing.html), which points to the collections definition.

**Pull request merged**